client-misk/src/main/kotlin/app/cash/backfila/client/misk/internal/BackfilaStartupConfigurator.kt

```diff
@@ -26,23 +22,34 @@ internal class BackfilaStartupConfigurator @Inject internal constructor(
   private val config: BackfilaClientConfig,
   private val backfilaClient: BackfilaClient,
   @ForBackfila private val moshi: Moshi,
-  @ForBackfila private val backfills: MutableMap<String, KClass<out Backfill<*, *, *>>>
+  private val backends: Set<BackfillOperator.Backend>
 ) : AbstractIdleService() {
   override fun startUp() {
     logger.info { "Backfila configurator starting" }
 
     val connectorDataAdapter = moshi.adapter<HttpConnectorData>()
     val httpConnectorData = HttpConnectorData(url = config.url)
 
+    // Build a list of registered Backfills from the different client backend implementations.
+    val registrations = mutableSetOf<BackfillOperator.BackfillRegistration>()
+    for (backend in backends) {
+      val backfills = backend.backfills()
+      // Make sure each backfill name is ony registered once in the service.
+      val intersection = registrations.map { it.name }.intersect(backfills.map { it.name })
+      require(intersection.isEmpty()) { "Found duplicate registrations: $intersection " }
+
+      registrations.addAll(backfills)
+    }
+
+    // Send the Backfila registration request
     val request = ConfigureServiceRequest.Builder()
         .backfills(
-            backfills.values.map { backfillClass ->
-              val description = (backfillClass.annotations.find { it is Description } as? Description)?.text
+            registrations.map { registration ->
               @Suppress("UNCHECKED_CAST")
-              val parameters = backfilaParametersForBackfill(backfillClass as KClass<Backfill<*, *, Any>>)
+              val parameters = backfilaParametersFromClass(registration.parametersClass)
```

Comment:
I actually wonder @shellderp whether the implementations should be in-charge of providing the whole parameters blob rather than just the class